src/wrappers/any.jl

```diff
@@ -20,21 +20,54 @@ function __init__()
     @initcxx
 end  # __init__()
 
-function node_manager_port()
+function parse_ray_args()
+   #==
+    "Starting agent process with command: ...
+    --node-ip-address=127.0.0.1 --metrics-export-port=60404 --dashboard-agent-port=60493
+    --listen-port=52365 --node-manager-port=58888
+    --object-store-name=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/sockets/plasma_store
+    --raylet-name=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/sockets/raylet
+    --temp-dir=/tmp/ray --session-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385
+    --runtime-env-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/runtime_resources
+    --log-dir=/tmp/ray/session_2023-08-14_14-54-36_055139_41385/logs
+    --logging-rotate-bytes=536870912 --logging-rotate-backup-count=5
+    --session-name=session_2023-08-14_14-54-36_055139_41385
+    --gcs-address=127.0.0.1:6379 --minimal --agent-id 470211272
+   ==#
     line = open("/tmp/ray/session_latest/logs/raylet.out") do io
         while !eof(io)
             line = readline(io)
-            if contains(line, "NodeManager server started")
+            if contains(line, "Starting agent process")
                 return line
             end
         end
     end
 
-    m = match(r"port (\d+)", line)
-    return m !== nothing ? parse(Int, m[1]) : error("Unable to find port")
+    gcs_match = match(r"gcs-address=(([0-9]{1,3}\.){3}[0-9]{1,3}:[0-9]{4})", line)
+    gcs_address = gcs_match !== nothing ? String(gcs_match[1]) : error("Unable to find GCS address")
+
+    node_ip_match = match(r"node-ip-address=(([0-9]{1,3}\.){3}[0-9]{1,3})", line)
+    node_ip = node_ip_match !== nothing ? String(node_ip_match[1]) : error("Unable to find Node IP address")
+
+    port_match = match(r"node-manager-port=([0-9]+)", line)
+    node_port = port_match !== nothing ? parse(Int, port_match[1]) : error("Unable to find Node Manager port")
+
+    return (node_ip, node_port, gcs_address)
 end
 
-initialize_coreworker() = initialize_coreworker(node_manager_port())
+
+function initialize_coreworker()
+
+    # TODO: are these defaults? can they be overwritten by user and/or Ray?
+    raylet_socket = "/tmp/ray/session_latest/sockets/raylet"
+    store_socket = "/tmp/ray/session_latest/sockets/plasma_store"
```

Comment:
specifically: are these deterministic irrespective of any runtime conditions such that we can set them as `const`? (TODO)